hasura/graphql-engine
# use readcommitted isolation level for event fetch

### Description

#6041 

### Changelog


### Affected components



### Steps to test and verify



`FOR UPDATE .. SKIP LOCKED` will enforce non-overlapping rows on concurrent event fetches and hence it is sufficient to run this in ReadCommitted. Read more here: https://www.postgresql.org/docs/9.5/sql-select.html#SQL-FOR-UPDATE-SHARE

Further, we do not expect the UPDATE to have any bad outcome in ReadCommitted with any other writes as it would either be a new INSERT (non-conflicting with fetched rows) or an UPDATE on a `locked=t` row (again non-conflicting with fetched rows). The only conflict that can happen is when the event trigger is deleted, in which case `archived=t` will be set and the event may or may not be delivered depending on which tx wins the lock first -- this is acceptable.
